### DISCUSSION TITLE
Nilan integration is taking long time to start

### DISCUSSION THREAD

**Discussion by Airler** [veista/nilan#106] [General]
Hi folks,
recently I noticed that the Nilan integration is taking a very long time to start up.
Based on Settings > System > Repairs > Integration startup time thie Nilan integration needs 1644.81s to start.
All other integrations need 25 or less seconds to start:
![image](https://github.com/veista/nilan/assets/17382580/5cbdc2f7-ed40-440b-bef8-dfd532782dbc)

I'm not sure if it is a problem with my setup or any other issue.
Does anyone else noticed this behaviour?

I have a Waveshare RS485 to PoE ETH (B) Modbus Gateway connected to my Nilan.

Kind regards,
Torben

**Comment by veista** [comment]
I do not at the moment. Are you running the latest versions of HA and the integration? I had similar issues a while ago.

After some update if HA couldn't connect on startup, it never did untill I reloaded the integration. That is why I added the auto-retry functionality. In the end it was a combination of an overloaded router/HA update.

**Reply by Airler** [reply]
I was using always the latest version of HA and Nilan integration.

After upgrading HA to 2024.1.0 on wednesday the issue disappeared and integration startup time was reduced to ~67s.
